src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1617,6 +1651,36 @@ def _openssh_public_key_bytes(self, key):
                 serialization._ssh_write_string(public_numbers.encode_point())
             )
 
+    def _parameter_bytes(self, encoding, format, cdata):
+        if encoding is serialization.Encoding.OpenSSH:
+            raise TypeError(
+                "OpenSSH encoding is not supported"
+            )
+
+        # Only DH is supported here currently.
+        q = self._ffi.new("BIGNUM **")
+        self._lib.DH_get0_pqg(cdata,
+                              self._ffi.NULL,
+                              q,
+                              self._ffi.NULL)
+        if encoding is serialization.Encoding.PEM:
+            if q[0] != self._ffi.NULL:
+                write_bio = self._lib.PEM_write_bio_DHxparams
```

Comment:
This is conditionally bound so it's possible for it to be unavailable on the lib object. Refresh my memory, is it possible to have a `DH *` with custom q when `DHxparams` doesn't exist? Also, what happens if you write non-custom-q DH params with `PEM_write_bio_DHxparams`? Does it explicitly write out the q or silently fall back to the behavior of `PEM_write_bio_DHparams`?